backend/app/helpers/spree/admin/base_helper.rb

```diff
@@ -115,6 +115,7 @@ def preference_fields(object, form)
              "form: f, attribute: \"preferred_\#{key}\", label: t(key, scope: 'spree') %>\n" \
         "<% end %>", deprecator: Spree::Deprecation
 
+      # @deprecated use link_to instead
```

Comment:
This won't really put a deprecation warning for users that are using this method in their code. 

We should start using the suggested versiona and add  a real deprecation message, something like:

```
def link_to_add_fields(name, target, options = {})
  name = '' if options[:no_text]
  css_classes = options[:class] ? options[:class] + " spree_add_fields" : "spree_add_fields"
  button_tag(name, data: { target: target }, class: css_classes)
end
deprecate link_to_add_fields: "Please use button_tag instead, Example: \n" \
  "button_tag \"Name\", data: { target: \"tbody#div\" }, deprecator: Spree::Deprecation
```

as we are doing for other methods in this file.

Reply:
ok. done. i had to add type: 'button' as a parameter to get it to work in all 3 contexts.